secure-api-gateway-ob-uk-common-error/src/main/java/com/forgerock/sapi/gateway/ob/uk/common/error/OBRIErrorType.java

```diff
@@ -158,11 +158,11 @@ public enum OBRIErrorType {
     PAYMENT_INVALID_INITIATION(
             HttpStatus.BAD_REQUEST,
             ErrorCode.OBRI_PAYMENT_INVALID,
-            "Payment invalid. Payment initiation received doesn't match the initial payment request"),
+            "Payment invalid. Payment initiation received doesn't match the initial payment request: '%s'"),
```

Comment:
Why did you do this? Is it to comply with a specific specification? 

Is there a danger here that we are leaking information that would make it easier for a hacker to work out how to make the request pass?

Reply:
The spec only define a bad request for that cases, I have added `%s` to add more information about the error to help the customer to identify the wrong property.
It doesn't leak any information that would make it easier for a hacker to make a request, only the name of the property that raise the error when doesn't match with the expected value.
The error description in the response will be something like:
```json
{
Payment invalid. Payment initiation received doesn't match the initial payment request: The property '${PROPERTY_NAME}' value does not match with the value provided in the consent
}
```

The same message that will be writing in the logs:
```shell
10:29:34.578 [main] ERROR c.f.s.g.o.u.r.s.a.o.p.s.v.RiskValidationService - The property 'MerchantCustomerIdentification' value does not match with the value provided in the consent
10:29:34.581 [main] ERROR c.f.s.g.o.u.r.s.a.o.p.s.v.RiskValidationService - The property 'PaymentPurposeCode' value does not match with the value provided in the consent
10:29:34.589 [main] ERROR c.f.s.g.o.u.r.s.a.o.p.s.v.RiskValidationService - The property 'DeliveryAddress' value does not match with the value provided in the consent
.....
```

Reply:
The log entry is fine as that is hopefully inaccessible to hackers. It could be considered that informing the caller which property differs helps them in their quest to form a correct request. 

What does the OBError1 object that gets returned via the API look like?

Reply:
Looks like:
**OBErrorResponse1**
```json
{
  "code": "OBRI.Request.Invalid",
  "id": "0e766963-ac24-4855-a1e6-cb666e6ee2d0",
  "message": "An error happened when parsing the request arguments",
  "errors": [
    {
      "errorCode": "OBRI.Payment.Invalid",
      "message": "Payment invalid. Payment risk received doesn't match the risk payment request: The property 'BeneficiaryAccountType' It was expected to be null as provided in the consent",
      "path": null,
      "url": null
    }
  ]
}
```

```json
{
  "code": "OBRI.Request.Invalid",
  "id": "fc9de982-9714-49a2-abeb-cea6e4c3d5ff",
  "message": "An error happened when parsing the request arguments",
  "errors": [
    {
      "errorCode": "OBRI.Payment.Invalid",
      "message": "Payment invalid. Payment risk received doesn't match the risk payment request: The property 'DeliveryAddress' value does not match with the value provided in the consent",
      "path": null,
      "url": null
    }
  ]
}
```

Reply:
OK. We can look at these later when we come up with our error framework. I feel that this should result in 
```

"errors": [
  {
    "errorCode": "UK.OBIE.Field.Invalid", 
    "message": "Payment invalid",
    "path": "Risk.DeliveryAddress", 
    "url": ""
  }
}
```
I guess we need to look at errors at some point. For now though I'm OK with it. Seems like a can of worms though!